### ISSUE TITLE
Split-from-training produces a Testing dataset whose train→test transition is only visible in data-flow lineage, not in Dataset_Type or dataset nesting

### ISSUE THREAD

**Issue by carlkesselman** [informatics-isi-edu/deriva-ml#272]
## Summary

When `split_dataset()` carves a labeled train/eval pair **out of an existing training dataset**, it produces a `Split` parent with two differently-used children — a `Training` child and a `Testing` child. That `Testing` child is 100% inside the project's training pool, but **nothing a consumer naturally dispatches on tells them so**:

- `Dataset_Type` gives the re-split's test child the **same** tags (`Testing` + `Labeled`) as a canonical held-out test set.
- The **dataset-nesting graph** is structurally identical for both a canonical split and a re-split-from-train (both are `Split → {Training, Testing}` roots).

The one place the distinction **is** recorded is **data-flow lineage**: the re-split's test child traces, through its producing split execution, back to a `Training` dataset. That train→test transition is the real, machine-checkable leakage signal — but it lives in `get_lineage`, which no type-dispatching consumer is prompted to consult.

> **Correction to the original filing:** an earlier draft of this issue claimed the train-pool origin "lives only in set-intersection on membership" and that "NFJ has no parent link to F3T." Both were imprecise. The link *does* exist — as the split execution's **consumed-dataset (input) edge** — and is cleanly retrievable via `get_lineage`. The accurate gap is that this edge is an *execution-level* edge, not a *dataset-nesting* edge, and neither `Dataset_Type` nor `list_dataset_relations` surfaces it.

## The two split families, side by side (catalog 2, verified)

**Canonical split** — test child drawn from the held-out test pool:
```
F3J [Split]  (root; .parents = [])
 ├── F3T [Training, Labeled]   550, from the TRAIN pool
 └── F44 [Testing,  Labeled]   550, from the TEST pool  ← genuinely held out
```

**Re-split-from-training** — test child drawn from the *training* pool:
```
NF0 [Split]  (root; .parents = [])
 ├── NF8 [Training, Labeled]   440, from F3T
 └── NFJ [Testing,  Labeled]   110, from F3T            ← 100% inside the train pool
```

`list_dataset_relations` shows the **same topology** for both (`Split → {Training, Testing}`, parent of the Split node empty). `Dataset_Type` is **identical** for F44 and NFJ. So neither the type nor the nesting graph distinguishes "held out from the canonical training pool" from "re-split out of the training pool."

### Where the distinction actually lives — data-flow lineage

`get_lineage(NFJ)` (verified):
```
NFJ (Testing)
  └─ producing_execution: NE0  "Create Labeled_Split and Small_Labeled_Split from the training set"
       └─ consumed_datasets: [ F3T  (CIFAR-10 training partition) ]   ← the train→test transition
```
Contrast `get_lineage(F44)`: its producing execution consumed the **test** pool, not F3T. **So the leakage IS machine-detectable** — the rule is "a `Testing` dataset whose lineage transitions through a `Training` dataset is not held out relative to that training set." Nothing today computes or surfaces that rule.

### Minimal repro

```python
from deriva_ml import DerivaML
ml = DerivaML(hostname="localhost", catalog_id="<cat>")

# (a) membership: the test child is entirely inside the train pool
def imgs(rid):
    return {m["RID"] for m in ml.lookup_dataset(rid).list_dataset_members().get("Image", [])}
F3T, F44, NFJ = imgs("F3T"), imgs("F44"), imgs("NFJ")
print(len(NFJ & F3T), len(NFJ & F44))   # -> 110 0   (all of NFJ is in the TRAIN pool)

# (b) the transition is in lineage, not in Dataset_Type or nesting:
#   get_lineage(NFJ) -> NFJ <- NE0 <- consumed F3T (Training)
#   list_dataset_relations(NFJ, recurse=True) -> parent NF0 (Split, root); never reaches F3T
```

## Why it matters

The platform's discipline (per the `dataset-lifecycle` skill) is "route on `Dataset_Type`, treat description as advisory." A modeler who trains on the full training partition (F3T/F38), then picks NFJ "because it's typed `Testing`," is evaluating on training images — silent leakage. The signal that would prevent it (the train→test lineage transition) is real and retrievable, but it's not the signal the consumer is told to use, and nothing flags the mismatch.

## Suggested directions (design decision for the platform — not prescriptive)

1. **Derive a leakage-aware signal from lineage.** Since the train→test transition is already in `get_lineage`, deriva-ml could expose a check (or a computed property / validation warning) of the form: *"this `Testing` dataset's lineage transitions through a `Training` dataset X — it is not held out relative to X."* This needs no schema change; it surfaces an edge that already exists.
2. **Promote the source edge to dataset nesting.** Have `split_dataset()` record the source training dataset as a **dataset-level parent** of the re-split's `Split` node (not just an execution input), so `list_dataset_relations(..., recurse=True)` climbs NFJ → NF0 → F3T. Makes the transition visible to consumers walking the dataset graph, not just data-flow lineage.
3. **A `Dataset_Type` qualifier** distinguishing canonical-holdout from re-split-from-train (e.g. a `Holdout` vs `Re-split` qualifier), so a pure type-dispatch consumer is correct without traversing any graph.
4. **At minimum, docs.** Warn in `split_dataset()` / `dataset-lifecycle` that splitting a *training* dataset yields a `Testing` child that is not held out relative to the original training pool.

(1) and (2) are the substantive options; (1) is lowest-touch since the lineage edge already exists.

## Provenance

Surfaced by the 2026-06-01 multipersona e2e run — Curator finding `labeled-test-splits-drawn-from-training-pool` (severity Medium), independently re-confirmed by the Analyst (lineage) and the cold evaluator (catalog parity). Graph topology + lineage re-verified directly against catalog 2 while triaging this issue.

🤖 Filed from the multipersona e2e evaluation.

**Comment by carlkesselman** [comment]
**Update — remediation direction moved to #273.**

After a design session (grill-with-docs, 2026-06-01) we concluded the *provenance-based* fixes proposed in this issue (a `Dataset_Type` qualifier, source→split dataset nesting, lineage-surfacing) are the wrong primary direction:

- They only catch leakage that happened *through a recorded split* — manual curation or re-import that produces overlap leaves no provenance edge, so the guard misses it.
- Source→split dataset nesting fights the deliberate ADR-0001 decision (lineage walks data-flow/execution edges, not dataset-nesting edges) and gets semantically messy (identity vs equivalence vs "symlink").

The agreed remediation is a **direct member-overlap guard** — test whether two datasets share element RIDs, history-independent, computed server-side. Design + live-verified efficiency in **#273**.

This issue stays open as the **finding of record** (the leakage trap is real and reproducible). The provenance options here remain valid alternatives, but #273 is the chosen path. Implementation of #273 is deferred.
